protocol/v1/qbft/instance/pre_prepare.go

```diff
@@ -95,15 +47,17 @@ upon receiving a valid ⟨PRE-PREPARE, λi, ri, value⟩ message m from leader(
 */
 func (i *Instance) UponPrePrepareMsg() pipelines.SignedMessagePipeline {
 	return pipelines.WrapFunc("upon pre-prepare msg", func(signedMessage *specqbft.SignedMessage) error {
-		prepareMsg, err := signedMessage.Message.GetProposalData()
-		if err != nil {
-			return errors.Wrap(err, "failed to get prepare message")
+		newRound := signedMessage.Message.Round
+
+		// A future justified proposal should bump us into future round and reset timer
+		if signedMessage.Message.Round > i.State().GetRound() {
+			i.ResetRoundTimer() // TODO: make sure what is needed here is i.ResetRoundTimer(), not something else
 		}
+		i.State().Round.Store(newRound)
```

Comment:
change this with `i.bumpToRound(newRound)` and move it to be inside the if above